shotgun_api3/shotgun.py

```diff
@@ -3539,6 +3526,137 @@ def _dict_to_extra_data(self, d, key_name="value"):
         """
         return dict([(k, {key_name: v}) for (k,v) in (d or {}).iteritems()])
 
+    def _upload_file_to_storage(self, path, storage_url):
+        """
+        Internal function to upload an entire file to the Cloud storage.
+        
+        :param str path: Full path to an existing non-empty file on disk to upload.
+        :param str storage_url: Target URL for the uploaded file. 
+        """
+        filename = os.path.basename(path)
+
+        fd = open(path, "rb")
+        try:
+            content_type = mimetypes.guess_type(filename)[0]
+            content_type = content_type or "application/octet-stream"
+            file_size = os.fstat(fd.fileno())[stat.ST_SIZE]
+            self._upload_data_to_storage(fd, content_type, file_size, storage_url )
+        finally:
+            fd.close()
+
+        LOG.debug("File uploaded to Cloud storage: %s", filename)
+
+    def _multipart_upload_file_to_storage(self, path, upload_info):
+        """
+        Internal function to upload a file to the Cloud storage in multiple parts.
+
+        :param str path: Full path to an existing non-empty file on disk to upload.
+        :param dict upload_info: Contains details received from the server, about the upload.
```

Comment:
would it helpful to document the expected contents of this dict, at least in one of the methods? It looks like we are accessing specific keys, so it's not just a case of sending it blindly back to the server. If we document it in one place there rest of the docstring could point to that one i guess....